tests/tests/account_data_service_tests.cpp

```diff
@@ -0,0 +1,119 @@
+#ifdef IS_TEST_NET
+#include <boost/test/unit_test.hpp>
+
+#include <scorum/chain/account_object.hpp>
+#include <scorum/chain/dbs_account.hpp>
+
+#include "../common/database_fixture.hpp"
+
+namespace scorum {
+namespace chain {
+
+private_key_type generate_private_key(const std::string& str)
+{
+    return private_key_type::regenerate(fc::sha256::hash(std::string(str)));
+}
+
+class create_account_with_data_service_fixture : public clean_database_fixture
+{
+public:
+    create_account_with_data_service_fixture()
+        : public_key(generate_private_key("user private key").get_public_key())
+        , data_service(db.obtain_service<dbs_account>())
+    {
+    }
+
+    void create_account()
+    {
+        data_service.create_account_by_faucets("user", "initdelegate", public_key, "", authority(), authority(),
+                                               authority(), asset(0, SCORUM_SYMBOL));
+    }
+
+    share_type calc_fee()
+    {
+        return std::max(db.get_witness_schedule_object().median_props.account_creation_fee.amount
+                            * SCORUM_CREATE_ACCOUNT_WITH_SCORUM_MODIFIER,
+                        share_type(100));
+    }
+
+    const public_key_type public_key;
+    dbs_account& data_service;
+};
+
+BOOST_FIXTURE_TEST_SUITE(create_account_with_data_service, create_account_with_data_service_fixture)
+
+BOOST_AUTO_TEST_CASE(check_account_name)
+{
+    try
+    {
+        create_account();
+
+        const account_object& acount = db.get_account("user");
+
+        BOOST_CHECK(acount.name == "user");
+    }
+    FC_LOG_AND_RETHROW()
```

Comment:
Why not just FC_CAPTURE_AND_LOG

Reply:
if there is exception test should fail, that's why we do rethrow

Reply:
I found only one usage of FC_CAPTURE_AND_LOG in the whole project.

Reply:
with FC_CAPTURE_AND_LOG output would be:
*** No errors detected